builders/build/canonical-contracts/precompiles/batch.md

```diff
@@ -0,0 +1,225 @@
+7. Press **transact**
+8. Press **Confirm** in the Metamask extension to confirm the transaction
+
+![Send Batch Transfer](/images/builders/build/canonical-contracts/precompiles/batch/batch-4.png)
+
+Once the transaction is complete, be sure to check both of the accounts' balances, either in Metamask or in a [block explorer](/builders/get-started/explorers/){target=_blank}. Congratulations! You've now sent a batched transfer via the Batch precompile.
+
+!!! note
+     Typically if you wanted to send the native currency to or through a contract, you would have to set the value within the overall transaction and interact with a payable function. However, since the Batch precompile interacts directly with Substrate code, this is not necessary.
+
+
+### Finding a Contract Interaction's Call Data {: #finding-a-contract-interactions-call-data } 
+
+Visual interfaces like Remix and handy libraries like ethers.js hide the way that ethereum transactions interact with solidity smart contracts. The name and input types of a function are hashed into a [function selector](https://docs.soliditylang.org/en/latest/abi-spec.html#function-selector-and-argument-encoding){target=_blank} and the input data is encoded. These two pieces are then combined and sent as the transaction's call data. To send a subtransaction within a batch transaction, the sender to know its call data beforehand. 
+
+Try finding a transaction's call data using Remix:
+
+1. Expand the SimpleMessage contract under **Deployed Contracts**
+2. Expand the **`setMessage`** function
+3. Enter a number and a message of your choice for **id** & **message**. In this example, **id** will be 1 and **message** will be "moonbeam"
+4. Instead of sending the transaction, click the copy button next to the **transact** button to copy the call data
+
+![Transaction Call Data](/images/builders/build/canonical-contracts/precompiles/batch/batch-5.png)
+
+Now you have the transaction's call data! Considering the example values of 1 and "moonbeam", we can keep an eye out for their encoded values in the call data:
+
+ --8<-- 'text/batch/simple-message-call-data.md'
+
+The calldata can be broken into 5 lines, where the first line is the function selector and the next four are the input data. 
+
+Note that the second line is equal to 1, which is the **id** that was provided. 
+
+What's left has to do with the **message** input. These last three lines are tricky, since strings are encoded with [RLP encoding](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/){target=_blank}. For sake of simplicity, understand that the third line refers to an offset, the fourth line refers to the string's length, and the fifth line is "moonbeam" in hexadecimal format with zeros for padding.
+
+### Function Interaction via Precompile {: #function-interaction-via-precompile }
```

Comment:
I think it is important to give the reader sort of like a summary of what you'll do in this example, for example "For this example, you'll be using the `batchAll` function to execute to contract calls in one single atomic transaction"